### PR TITLE
Remove JDBC KVS

### PR DESCRIPTION
**Pull request by pkoenig10**
Fixes https://github.com/palantir/atlasdb/issues/1198

For motivation, see https://github.com/palantir/atlasdb/pull/6098#issuecomment-1167960268.

This PR also removes an unnecessary RocksDB dependency from `atlasdb-client` and `atlasdb-config`. RocksDB is only used in tests and those classes have been moved to `atlasdb-test-shared`.

### CODE REVIEW

**Review by jeremyk-91 — COMMENTED** [review]
I'm mostly confident this change does what it purports to do.

Some questions:
- Would it be feasible to test that this would not break the large internal product if we were to bump their version of AtlasDB to this version?
- How did we verify that no one is still using JDBC? I think I searched for `type: jdbc` in yaml files and had a brief look around, but curious if there's a better way to do that

Other than that there are a bunch of nits though this broadly seems fine.

File: atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java [hunk 1]
```diff
@@ -75,9 +75,9 @@ public static void setUp() {
         when(CASSANDRA_CONFIG_WITHOUT_NAMESPACE.namespace()).thenReturn(Optional.empty());
         when(CASSANDRA_CONFIG_WITH_NAMESPACE.namespace()).thenReturn(Optional.of(TEST_NAMESPACE));
 
-        when(KVS_CONFIG_WITHOUT_NAMESPACE.type()).thenReturn("rocksdb");
-        when(KVS_CONFIG_WITH_OTHER_NAMESPACE.type()).thenReturn("database");
-        when(KVS_CONFIG_WITH_NAMESPACE.type()).thenReturn("sqlite");
+        when(KVS_CONFIG_WITHOUT_NAMESPACE.type()).thenReturn("type");
+        when(KVS_CONFIG_WITH_OTHER_NAMESPACE.type()).thenReturn("type");
+        when(KVS_CONFIG_WITH_NAMESPACE.type()).thenReturn("type");
```
[the remark this answers is not in the document]
Comment (pkoenig10) [reply atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java:80]: No, this was an unrelated cleanup. It felt a bit weird to use values that appeared to be legitimate types but are not.

File: docs/source/services/key_value_services/index.rst [hunk 2]
```diff
@@ -67,24 +55,3 @@ DB KVS
...
---------
-
-JDBC KVS existed prior to DB KVS and was the primary means for deploying applications on Postgres.
-JDBC KVS has been deprecated in favor of DB KVS.
-
-Deleted
-=======
-
-ClusterKeyValueService.java - Intended to be run on top of a MySQL/Cluster
-RocksDbKVS - Removed in 0.44
```
[the remark this answers is not in the document]
Comment (pkoenig10) [reply docs/source/services/key_value_services/index.rst:90]: > the CQL KVS still exists, right?

I couldn't find anything by looking at `KeyValueService` subclasses. But maybe I wasn't looking in the right place?

**Comment by pkoenig10** [comment]
> Would it be feasible to test that this would not break the large internal product if we were to bump their version of AtlasDB to this version?

Sure, is there a way to publish snapshots that can be used there?

> How did we verify that no one is still using JDBC? I think I searched for type: jdbc in yaml files and had a brief look around, but curious if there's a better way to do that

Yeah I just did a code search.

**Comment by pkoenig10** [comment]
I also performed an internal search for `atlasdb-jdbc` in `versions.lock` files. I only found 8 occurrences in production services, all of which are Foundry services which I am confident do not use the JDBC KVS.

File: docs/source/services/key_value_services/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply docs/source/services/key_value_services/index.rst:90]: womp, my bad!

File: atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply atlasdb-config/src/test/java/com/palantir/atlasdb/config/AtlasDbConfigTest.java:80]: Fair enough. Yeah that's ok

**Comment by jeremyk-91** [comment]
tagged `0.637.0-rc1`. Thanks!

**Comment by jeremyk-91** [comment]
oops, it already exists :(

**Comment by jeremyk-91** [comment]
could we rebase the branch on develop? happy to help with this if need be / if you don't have bandwidth!

**Comment by pkoenig10** [comment]
Looks like CI failed. Can you trigger a re-run?

https://app.circleci.com/pipelines/github/palantir/atlasdb/9039/workflows/2d89f644-8d25-45c4-a214-404f5909f37b

**Review by jeremyk-91 — APPROVED** [review]
👍 

Discussed offline: we were able to get large internal product to compile. There were some _unrelated_ runtime failures relating to #6018, but there were other ETE tests that passed, and given that I don't anticipate this PR to cause runtime only failures, I think we should be good.

**Comment by jeremyk-91** [comment]
Administrative Permissions: My only commit is an Autorelease.

**Comment by pkoenig10** [comment]
👍

**Pull request merged**